Eliminated no-type/no-original TypeTrees.

paulp opened the pull request:
Noticed many calls reaching a spot which they should not.
Added a devWarning so the next guy might find it more quickly.

adriaanm commented:
PLS REBUILD ALL
NOLITTER!

xeno-by commented on src/compiler/scala/tools/nsc/typechecker/Typers.scala:
```diff
@@ -5241,11 +5241,13 @@ trait Typers extends Adaptations with Tags with TypersTracking {
           // and we try again (@see tryTypedApply). In that case we can assign
           // whatever type to tree; we just have to survive until a real error message is issued.
+          devWarning(tree.pos, s"TypeTree reached typedTypeTree with null original. Assigning Any in desperation.")
           tree setType AnyTpe
```
I wonder what was the motivation for this.

paulp replied:
I'm not sure if I understand the question (the original author's motivation?) but as for me, the comment says approximately "compilation has failed, we'll pitch an Any into the mix and an error will be reported." And I see it walking that code path 1500 times during a successful compilation. And when one is trying to make the pattern matcher a tiny bit sound one starts seeing the random introduction of "Any" types as deadly poison.

xeno-by replied:
Yeah, the motivation of `tree setType AnyTpe`. And thanks for explaining the background for your change!

scala-jenkins commented:
(kitty-note-to-self: ignore 22348197)
:cat: Roger! Rebuilding pr-scala for 941de781. :rotating_light:

scala-jenkins commented:
(kitty-note-to-self: ignore 22348197)
:cat: cleaning up... sorry! :cat:

paulp commented:
I eventually caught a glimpse through the noise that this actually doesn't compile. (I take it I'm not the only one for whom that was not self-apparent.) That'll teach me to try to carve a little patch off a big patch.

adriaanm commented:
That's a bug in the scripts. Looking into it.

adriaanm commented:
"-e" missing on bash invocation

adriaanm commented:
Should be fixed by https://github.com/scala/jenkins-scripts/commit/b7d564b976bebcaed55e42ba3bb747e455078f02

adriaanm commented:
Please reopen. Trying to remove red from queue.

paulp commented:
Updated with less broken cherrypick.

paulp commented:
PLS REBUILD ALL
(Not sure if I have to do that)

scala-jenkins commented:
(kitty-note-to-self: ignore 22530824)
:cat: Roger! Rebuilding pr-scala for 02df8f3f, 466b7d29, e7876d5c, b145cb1c, b741e8ad, e5c99b1b, 8589deb4, 651e1988, e3606bf1. :rotating_light:

adriaanm commented:
No need for an explicit rebuild if the commits are new.

adriaanm commented:
It looks like something went wrong with your rebase.

paulp committed:
Eliminated no-type/no-original TypeTrees.

Noticed many calls reaching a spot which they should not.
Added a devWarning so the next guy might find it more quickly.

paulp commented:
It's awfully challenging to find the part which suggests what the problem was. I took a stab.
